fields/types/relationship/RelationshipType.js

```diff
@@ -232,9 +237,9 @@ relationship.prototype.updateItem = function (item, data, callback) {
 			item.set(this.path, _new);
 		}
 	} else {
-		if (data[this.path]) {
-			if (data[this.path] !== item.get(this.path)) {
-				item.set(this.path, data[this.path]);
+		if (value) {
```

Comment:
We don't want the 'else if' to run if there is a value though the if statements should likely be tidied up to make that clear